File: app/schemas/project_info.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ProjectMetadata(BaseModel):
    name: str
    description: str
    github_url: str
    contact_email: str
    tutorial_available: bool = False


class ProjectStatistics(BaseModel):
    provider_count: int = Field(ge=0)
    model_count: int = Field(ge=0)
    prompt_count: int = Field(ge=0)
    prompt_version_count: int = Field(ge=0)
    test_task_count: int = Field(ge=0)
    test_unit_count: int = Field(ge=0)
    test_experiment_count: int = Field(ge=0)


class ProjectUpdateGuidance(BaseModel):
    deployment_type: str
    title: str
    steps: list[str]
    commands: list[str] = Field(default_factory=list)


class ProjectVersionInfo(BaseModel):
    current: str
    latest: str | None = None
    has_update: bool = False
    check_status: str = "unknown"
    check_error: str | None = None
    release_url: str | None = None
    deployment_type: str
    update_guidance: ProjectUpdateGuidance


class ProjectInfoSummary(BaseModel):
    project: ProjectMetadata
    version: ProjectVersionInfo
    statistics: ProjectStatistics

File: frontend/src/api/projectInfo.ts
import { request } from './http'

export interface ProjectMetadataResponse {
  name: string
  description: string
  github_url: string
  contact_email: string
  tutorial_available: boolean
}

export interface ProjectStatisticsResponse {
  provider_count: number
  model_count: number
  prompt_count: number
  prompt_version_count: number
  test_task_count: number
  test_unit_count: number
  test_experiment_count: number
}

export interface ProjectUpdateGuidanceResponse {
  deployment_type: string
  title: string
  steps: string[]
}

export interface ProjectVersionInfoResponse {
  current: string
  latest: string | null
  has_update: boolean
  release_url: string | null
  deployment_type: string
  update_guidance: ProjectUpdateGuidanceResponse
}

export interface ProjectInfoSummaryResponse {
  project: ProjectMetadataResponse
  version: ProjectVersionInfoResponse
  statistics: ProjectStatisticsResponse
}

export function getProjectInfoSummary(): Promise<ProjectInfoSummaryResponse> {
  return request<ProjectInfoSummaryResponse>('/project-info/summary')
}

export function checkProjectVersion(): Promise<ProjectVersionInfoResponse> {
  return request<ProjectVersionInfoResponse>('/project-info/version/check')
}
